tests/test_composite.py

```diff
@@ -123,7 +123,7 @@ def test_multi_block_set_get_ers():
     assert multi.n_blocks == 6  # Check pyvista side registered it
     # Add data to the MultiBlock
     data = ex.load_rectilinear()
-    multi[1, 'rect'] = data
+    multi[1] = ('rect', data)
```

Comment:
This still seems weird to me, but I don’t know how to address this unless we make MultiBlock either behave as a dict or a list. For the time being this works, and we may not be able to come up with a better solution.

Reply:
I'm okay getting rid of this entirely.  I think 95% or more use cases will use `append` which now has a nice API.  This use case is for when someone wants to overwrite in the middle of the multiblock, and set the name simultaneously.  Without this functionality, it only takes one more line to do this, i.e.:

```py
multi[2] = mesh
multi.set_block_name(2, "name")
```

Reply:
I’d prefer this. From my brief usage of multiblock, I’d agree that most of the time you’re either appending or overwriting.

Reply:
I took 'this' as removing the tuple assignment.  This feels much better to me, and cements the API as being list-like, but with dict-bells-and-whistles when possible.